src/utils/rabbitmq.js

```diff
@@ -1,21 +1,129 @@
 const { Configuration } = require('@schul-cloud/commons');
 const amqp = require('amqplib');
+const logger = require('../logger');
 
 let connection;
 
-const createChannel = async () => {
+const getConnection = async () => {
+	if (!Configuration.get('FEATURE_RABBITMQ_ENABLED') || !Configuration.get('RABBITMQ_URI')) {
+		throw new Error('RabbitMQ is not configured.');
+	}
+
 	if (connection) {
-		const con = await connection; // ensure connection has resolved
-		return con.createChannel();
+		return connection;
 	}
 
-	throw new Error('No RabbitMQ connection available.');
+	try {
+		connection = await amqp.connect(Configuration.get('RABBITMQ_URI'));
+		connection.on('close', () => {
+			logger.warning('RabbitMQ connection was closed.');
+			connection = undefined;
+		});
+		return connection;
+	} catch (err) {
+		connection = undefined;
+		throw err;
+	}
 };
 
-const setup = async (app) => {
+const setup = async () => {
 	if (Configuration.get('FEATURE_RABBITMQ_ENABLED')) {
-		connection = amqp.connect(Configuration.get('RABBITMQ_URI'));
+		try {
+			await getConnection();
+		} catch (err) {
+			logger.error('RabbitMQ unable to setup.', err);
+		}
 	}
 };
 
-module.exports = { setup, createChannel };
+class Channel {
+	constructor(queue, queueOptions) {
+		this.queue = queue;
+		this.queueOptions = queueOptions || {};
+	}
+
+	async connect() {
+		try {
+			const conn = await getConnection();
+			this.channel = await conn.createChannel();
+			this.channel.assertQueue(this.queue, this.queueOptions);
+			this.channel.on('close', () => {
+				logger.warning('RabbitMQ channel was closed.');
+				this.channel = undefined;
+			});
+			return this.channel;
+		} catch (err) {
+			this.channel = undefined;
+			throw err;
```

Comment:
Where will this exception catched?

Reply:
see other [comment](https://github.com/hpi-schul-cloud/schulcloud-server/pull/1761/files/52d0cbd9a4edf068210945e781d6d6dffbd1a167#r492166261)